examples/registration.rb

```diff
@@ -0,0 +1,73 @@
+# This program provides a simple example illustrating how to programmatically
+# register a client with InnoVault and e3db. In some situations, it's preferable
+# to register a client from the server or system that will be using its
+# credentials (to ensure that all data is truly encrypted from end-to-end
+# with no possibilities of a credential leak). For more detailed information,
+# please see the documentation home page: https://tozny.com/documentation/e3db
+#
+# Author::    Eric Mann (eric@tozny.com)
+# Copyright:: Copyright (c) 2017 Tozny, LLC
+# License::   Public Domain
+
+# ---------------------------------------------------------
+# Initialization
+# ---------------------------------------------------------
+
+require 'e3db'
+
+# A registration token is required to set up a client. In this situation,
+# we assume an environment variable called REGISTRATION_TOKEN is set
+token = ENV["REGISTRATION_TOKEN"]
+
+# Clients can either create new cryptographic keypairs, or load in a pre-defined
+# pair of Curve25519 keys. In this situation, we will generate a new keypair.
+private_key, public_key = E3DB::Client.generate_keypair
+
+puts("Public Key:  " + public_key)
+puts("Private Key: " + private_key)
+
+# The e3db server keeps track of the name of the curve used with public keys,
+# so we need to wrap the generated version with an object helper
+wrapped_key = E3DB::PublicKey.new(:curve25519 => public_key)
```

Comment:
It seems like registration should handle this wrapping? I don't think we're aiming for flexibility (and shouldn't). And we don't support other curves atm, anyway.